packages/android_intent/example/test_driver/android_intent_e2e.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:e2e/e2e.dart';
+
+import 'package:android_intent_example/main.dart';
+
+/// This is a smoke test that verifies that the example app builds and loads.
+/// Because this plugin works by launching Android platform UIs it's not
+/// possible to meaningfully test it through its Dart interface currently. There
+/// are more useful unit tests for the platform logic under android/src/test/.
+void main() {
+  E2EWidgetsFlutterBinding.ensureInitialized();
+  testWidgets('Embedding example app loads', (WidgetTester tester) async {
+    // Build our app and trigger a frame.
+    await tester.pumpWidget(MyApp());
+
+    if (Platform.isAndroid) {
+      expect(
+        find.byWidgetPredicate(
+          (Widget widget) =>
+              widget is Text && widget.data.startsWith('Tap here'),
+        ),
+        findsNWidgets(2),
+      );
+    } else {
+      expect(
```

Comment:
I wish we had a simple way to actually make sure that the plugin is registered (can we fire an "invalid" intent? and hope that the app is still in the foreground for the test to pass?)

Reply:
I could just call `launch()`, similar to [url_launcher_e2e](https://github.com/flutter/plugins/blob/master/packages/url_launcher/url_launcher/example/test_driver/url_launcher_e2e.dart), but I didn't originally because I don't think there's anything meaningful to assert for. The API doesn't return the intent result at all. I could wait for the future to resolve at least and make sure that it doesn't throw or hang...? I don't know if that's really any more meaningful than this smoke test that the example app's UI loads.

Reply:
Yeah if it doesn't throw it means the plugin is registered, so we'd be testing that the plugin is working with the new embedder (e.g right now the test can pass even if the plugin isn't registered)

Reply:
I added another test for Android asserting that a method was called and threw with an "Activity not found" exception. I think that's a pretty reasonable smoke test that the plugin is registered.